crates/shim/src/lib.rs

```diff
@@ -239,3 +261,34 @@ pub fn run(callback: impl Fn(Task) -> Result<TaskResult>) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(not(feature = "vsock"))]
+pub fn run(callback: impl Fn(Task) -> Result<TaskResult>) -> Result<()> {
+    let workspace = gevulot_common::WORKSPACE_PATH;
+
+    println!("waiting for {workspace} mount to be present");
+    let beginning = Instant::now();
+    loop {
+        if beginning.elapsed() > MOUNT_TIMEOUT {
+            panic!("{} mount timeout", workspace);
+        }
+
+        if mount_present(workspace)? {
+            println!("{workspace} mount is now present");
+            break;
+        }
+
+        sleep(Duration::from_secs(1));
+    }
+
+    let file = File::open(PathBuf::from(WORKSPACE_PATH).join(TASK_FILE_NAME))?;
+    let task: Task = serde_json::from_reader(file)?;
+
+    let result = callback(task).map_err(|e| e.to_string());
+    let mut result_file =
+        File::create_new(PathBuf::from(WORKSPACE_PATH).join(TASK_RESULT_FILE_NAME))?;
```

Comment:
While trying to integrate Aztec's ACVM - [based](https://github.com/AztecProtocol/aztec-packages/blob/master/noir/noir-repo/rust-toolchain.toml) on stable Rust `v1.74.1` - I got the following error:

```
error[E0658]: use of unstable library feature 'file_create_new'
   --> /home/andrew/.cargo/git/checkouts/gevulot-cd67166d40edf61d/ad0cd45/crates/shim/src/lib.rs:289:9
    |
289 |         File::create_new(PathBuf::from(WORKSPACE_PATH).join(TASK_RESULT_FILE_NAME))?;
    |         ^^^^^^^^^^^^^^^^
    |
    = note: see issue #105135 <https://github.com/rust-lang/rust/issues/105135> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `gevulot-shim` (lib) due to previous error
```

The issue is that `Flie::create_new` got stabilized quite recently - it is [available](https://github.com/rust-lang/rust/blob/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/std/src/fs.rs#L436-L439) since Rust 1.77, which was released ~5 months ago.

@tuommaki What about replacing this line with `OpenOptions::new().read(true).write(true).create_new(true).open(path.as_ref())`, so we can support older projects?

Reply:
Yeah, good shout! I actually hit this problem during the weekend as well so changed it now :+1: